### PR TITLE
Swift3

### PR DESCRIPTION
**Pull request by mitulmanish**
Migration from Swift 2.3 to 3.0

### CODE REVIEW

**Comment by tatey** [comment]
Hey @mitulmanish. This looks great. Would love to see it merged back upstream. Open a PR and I'll merge.

**Comment by mitulmanish** [comment]
Hi @tatey .Nice to hear from you.I am currently making some more changes and getting it reviewed from @flowsprenger .Will create a PR for upstream soon.

**Comment by tatey** [comment]
Great! Look forward to it 😄

File: Source/Client.swift [hunk 1]
```diff
@@ -9,128 +9,129 @@ public class Client {
...
-
-	public func lightTargetWithSelector(selector: LightTargetSelector) -> LightTarget {
+	
+	public func lightTargetWithSelector(_ selector: LightTargetSelector) -> LightTarget {
 		return LightTarget(client: self, selector: selector, filter: selectorToFilter(selector))
 	}
-
-	func addObserver(lightsDidUpdateHandler: ClientObserver.LightsDidUpdate) -> ClientObserver {
+	
+	func addObserver(lightsDidUpdateHandler: @escaping ClientObserver.LightsDidUpdate) -> ClientObserver {
```
Comment (flowsprenger) [inline Source/Client.swift]: lightsDidUpdateHandler should be optional to avoid breaking clients

File: Source/Client.swift [hunk 2]
```diff
@@ -9,128 +9,129 @@ public class Client {
...
 		let observer = ClientObserver(lightsDidUpdateHandler: lightsDidUpdateHandler)
 		observers.append(observer)
 		return observer
 	}
-
+	
 	func removeObserver(observer: ClientObserver) {
```
Comment (flowsprenger) [inline Source/Client.swift]: observer should be optional to avoid breaking clients

File: Source/ClientObserver.swift [hunk 3]
```diff
@@ -6,11 +6,11 @@
...
 class ClientObserver {
-	typealias LightsDidUpdate = (lights: [Light]) -> Void
-
+	typealias LightsDidUpdate = (_ lights: [Light]) -> Void
+	
 	let lightsDidUpdateHandler: LightsDidUpdate
-
-	init(lightsDidUpdateHandler: LightsDidUpdate) {
+	
+	init(lightsDidUpdateHandler: @escaping LightsDidUpdate) {
```
Comment (flowsprenger) [inline Source/ClientObserver.swift]: lightsDidUpdateHandler should be optional to avoid breaking clients

File: Source/LightTargetObserver.swift [hunk 4]
```diff
@@ -7,10 +7,10 @@ import Foundation
 
 public class LightTargetObserver {
 	public typealias StateDidUpdate = () -> Void
-
+	
 	let stateDidUpdateHandler: StateDidUpdate
-
-	init(stateDidUpdateHandler: StateDidUpdate) {
+	
+	init(stateDidUpdateHandler: @escaping StateDidUpdate) {
```
Comment (flowsprenger) [inline Source/LightTargetObserver.swift]: stateDidUpdateHandler should be optional to avoid breaking clients

**Pull request merged**